nsv13/code/modules/overmap/knpc.dm

```diff
@@ -67,114 +67,112 @@ GLOBAL_LIST_EMPTY(knpcs)
 	if(get_dist(H, their_id.loc) > 1)
 		return FALSE
 	var/obj/item/card/id/ID = H.get_idcard()
-	if(ID && istype(ID, /obj/item/card/id/syndicate) || istype(ID, /obj/item/card/id/syndicate_command) || istype(ID, /obj/item/card/id/syndi_crew))
-		their_id.forceMove(get_turf(H))
-		H.visible_message("<span class='warning'>[H] snatches [their_id]. </span>")
-		ID.access |= their_id.access
-	if(!ID)
+	if(ID)
+		if(istype(ID, /obj/item/card/id/syndicate) || istype(ID, /obj/item/card/id/syndicate_command) || istype(ID, /obj/item/card/id/syndi_crew))
+			their_id.forceMove(get_turf(H))
+			H.visible_message("<span class='warning'>[H] snatches [their_id]. </span>")
+			ID.access |= their_id.access
+	else
 		H.put_in_inactive_hand(their_id)
 		if(H.equip_to_appropriate_slot(their_id))
 			H.update_inv_hands()
 			return TRUE
 
 /datum/component/knpc/Destroy(force, silent)
 	GLOB.knpcs -= src
-	. = ..()
+	return ..()
 
-/datum/component/knpc/proc/pathfind_to(atom/movable/target, turf/avoid)
+/datum/component/knpc/proc/pathfind_to(atom/target, turf/avoid)
 	var/mob/living/carbon/human/ai_boarder/H = parent
-	if(dest && dest == get_turf(target) || H.stat == DEAD || H.incapacitated())
-		return FALSE //No need to recalculate this path.
-	path = list()
-	dest = null
-	var/obj/item/card/id/access_card = H.wear_id
-	if(target)
-		dest = get_turf(target)
-		path = get_path_to(H, dest, /turf/proc/Distance_cardinal, 0, 120, id=access_card, exclude=avoid, simulated_only=!(H.wear_suit?.clothing_flags & STOPSPRESSUREDAMAGE && H.head?.clothing_flags & STOPSPRESSUREDAMAGE))
-
-		var/obj/structure/dense_object = locate(/obj/structure) in get_turf(get_step(H, H.dir)) //If we're stuck
-		if(istype(dense_object, /obj/structure/table) || istype(dense_object, /obj/structure/railing))
-			H.forceMove(get_turf(dense_object))
-			H.visible_message("<span class='warning'>[H] climbs onto [dense_object]!</span>")
-			H.Stun(2 SECONDS) //Table.
-		var/obj/machinery/door/firedoor/border_only/fuckingMonsterMos = locate(/obj/machinery/door/firedoor/border_only) in get_turf(get_step(H, H.dir))
-		if(fuckingMonsterMos && istype(fuckingMonsterMos))
-			fuckingMonsterMos.open()
+	var/turf/targturf = get_turf(target)
+	if(dest && dest == targturf || H.incapacitated())
+		return FALSE //No need to recalculate the path.
+	dest = targturf
+	var/turf/T = get_step(H, H.dir)
+	var/obj/structure/dense_object = locate(/obj/structure) in T //If we're stuck
+	if(climbable[dense_object.type])
```

Comment:
getting a runtime here in testing - cannot read null.type
none of them will move anywhere.